lib/ansible/modules/cloud/azure/azure_rm_trafficmanagerprofile.py

```diff
@@ -0,0 +1,551 @@
+                    # create a new profile. Return a profile object to the user
+                    if not bool(current_traffic_manager):
+                        self.results['state'] = results
+                    # if the profile exists, compare and report back the differences
+                    else:
+                        self.results['state'] = self.compare_profiles(
+                            current_traffic_manager,
+                            self.properties)
+                    return self.results
+
+                # Not in check_mode. This will create the profile.
+                else:
+                    if not bool(current_traffic_manager):
+                        # profile doesn't exist. We will created it.
+                        # Traffic Manager uses DNS, we need to check if the
+                        # name used in the DNS prefix is available
+                        try:
+                            dns_available = dict()
+                            dns_available = self.trafficmanager_client.profiles.check_traffic_manager_relative_dns_name_availability(
+                                self.name, 'Microsoft.Network/trafficManagerProfiles')
+                            if bool(dns_available.name_available):
+                                pass
+
+                        except CloudError as cloud_error:
+                            self.fail('Error checking the traffic manager DNS: {0}. {1}'.format(self.name, str(cloud_error)))
+
+                        traffic_manager = self.create_or_update_traffic_manager_profile(
+                            self.resource_group,
+                            self.name,
+                            self.location,
+                            self.properties)
+
+                    results = self.traffic_manager_to_dict(
+                        traffic_manager,
+                        self.resource_group,
+                        self.name,
+                        self.location,
+                        self.properties
+                    )
+                    changed = True
+
+            elif self.state == 'absent':
+                self.log('Deleting traffic manager {0}'.format(self.name))
+                # dry-run. report back to the user.
+                if self.check_mode:
+                    self.results['state'] = dict(
+                        resouce_group=self.resource_group,
+                        name=self.name)
+                    return self.results
+                if bool(current_traffic_manager):
+                    # Deletes the traffic manager and set change variable
+                    self.remove_traffic_manager_profile(
+                        self.resource_group,
+                        self.name)
+                    changed = True
+        except CloudError:
+            if self.state == 'present':
+                changed = True
+
+        self.results['changed'] = changed
+        self.results['state'] = results
```

Comment:
The `self.results['state']` is already set in the previous complicated logic, why here re-assign?